### PR TITLE
Privacy

### PR DESCRIPTION
**Pull request by CastorK**
Made it so that bookables can be private or public. You can view the calendar of a public bookable even if you're not logged in, private bookables can not be. If you try to view a private bookable without being logged in you're redirected to a login page

### CODE REVIEW

File: fars/booking/models.py [hunk 1]
```diff
@@ -11,6 +11,7 @@ class Bookable(models.Model):
     name = models.CharField(max_length=64)
     description = models.CharField(max_length=256)
     icon = models.CharField(max_length=32, default='tf.svg')
+    public = models.BooleanField(default=False)
```
Comment (tlangens) [inline fars/booking/models.py]: I'd prefer this to be "visibility" as some enum, so you can have different degrees of visibility, e.g. public, internal, private etc

**Comment by CastorK** [comment]
Added some admin functionality. Now each bookable gets its own admin group which superusers can add people to to make them admins of that bookable. Superusers are still admins of all bookables without having to be in all groups

File: fars/booking/templates/login.html [hunk 2]
```diff
@@ -3,32 +3,39 @@
 
 {% block css %}
   <link rel="stylesheet" href="{% static 'css/calendar-theme.css' %}">
+  <style>
```
Comment (tlangens) [inline fars/booking/templates/login.html]: is there a reason to not just create login.css?

File: fars/booking/views.py [hunk 3]
```diff
@@ -16,21 +17,27 @@ def home(request):
...
     bookable_obj = get_object_or_404(Bookable, id_str=bookable)
-    context = {
-        'date': "{y}-{m:02d}-{d:02d}".format(y=year, m=month, d=day),
-        'bookable': bookable_obj,
-        'user': request.user
-    }
-    return render(request, 'day.html', context)
+    if not bookable_obj.public and not request.user.is_authenticated:
+        return redirect('{}?next={}'.format(reverse('login'), request.path_info))
+    else:
```
Comment (tlangens) [inline fars/booking/views.py]: If you return inside if, then else is not necessary. Makes code a bit leaner.

File: fars/booking/views.py [hunk 4]
```diff
@@ -16,21 +17,27 @@ def home(request):
...
 def bookings_month(request, bookable):
     bookable_obj = get_object_or_404(Bookable, id_str=bookable)
-    context = {
-        'bookable': bookable_obj,
-        'user': request.user
-    }
-    return render(request, 'month.html', context)
+    if not bookable_obj.public and not request.user.is_authenticated:
+        return redirect('{}?next={}'.format(reverse('login'), request.path_info))
+    else:
```
Comment (tlangens) [inline fars/booking/views.py]: see comment below.

File: fars/booking/templatetags/tags.py [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
 @register.filter
 def modulo(num, val):
     return num % val
+
+@register.filter
+def is_bookablestaff(user, bookable):
+    groupname = "{}_admin".format(bookable.id_str)
+    groupmembers = User.objects.filter(groups__name=groupname)
+    
+    if user in groupmembers:
```
Comment (tlangens) [inline fars/booking/templatetags/tags.py]: I think
```
if user in groupmembers or user.is_superuser:
    return True
return False
```
is more readable

File: fars/booking/templatetags/tags.py [hunk 6]
```diff
@@ -1,7 +1,18 @@
...
+from django.contrib.auth.models import User
 
 register = template.Library()
 
 @register.filter
 def modulo(num, val):
     return num % val
+
+@register.filter
+def is_bookablestaff(user, bookable):
```
Comment (tlangens) [inline fars/booking/templatetags/tags.py]: If you renamed group name to admin then this should be admin too.

**Comment by tlangens** [comment]
👌

**Pull request merged**